.pipelines/templates/role-assignment.yaml

```diff
@@ -0,0 +1,31 @@
+steps:
+  - script: |
+      ASSIGNEE_OBJECT_ID="$(az identity show -g $(CLUSTER_RESOURCE_GROUP) -n ${AZURE_CLUSTER_NAME}-agentpool --query principalId -otsv)"
+      echo "ASSIGNEE_OBJECT_ID='${ASSIGNEE_OBJECT_ID}'"
+      echo "##vso[task.setvariable variable=ASSIGNEE_OBJECT_ID]${ASSIGNEE_OBJECT_ID}"
+
+      ROLE_ASSIGNMENT_IDS=""
+
+      az role assignment create --assignee-object-id "${ASSIGNEE_OBJECT_ID}" --assignee-principal-type "ServicePrincipal" --role "Virtual Machine Contributor" --scope "/subscriptions/$(SUBSCRIPTION_ID)/resourcegroups/$(CLUSTER_RESOURCE_GROUP)"
+      az role assignment create --assignee-object-id "${ASSIGNEE_OBJECT_ID}" --assignee-principal-type "ServicePrincipal" --role "Managed Identity Operator" --scope "/subscriptions/$(SUBSCRIPTION_ID)/resourcegroups/$(CLUSTER_RESOURCE_GROUP)"
+
+      # permission for Key Vault
+      if [[ -n "$(USER_MSI_RESOURCE_GROUP)" ]]; then
+        ID="$(az role assignment create --assignee-object-id "${ASSIGNEE_OBJECT_ID}" --assignee-principal-type "ServicePrincipal" --role "Reader" --scope "/subscriptions/$(SUBSCRIPTION_ID)/resourcegroups/$(USER_MSI_RESOURCE_GROUP)/providers/Microsoft.KeyVault/vaults/$(KEYVAULT_NAME)" --query id -otsv)"
```

Comment:
The CSI driver doesn't require this permission on the keyvault. Do we need this?

Reply:
yup. removed it. [build](https://dev.azure.com/AzureContainerUpstream/Secrets%20Store%20CSI%20Driver%20Provider%20Azure/_build/results?buildId=26667&view=results) is passing.